pkg/controllers/dynakube/activegate/internal/statefulset/builder/modifiers/eec.go

```diff
@@ -0,0 +1,82 @@
+package modifiers
+
+import (
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/activegate/consts"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/activegate/internal/statefulset/builder"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/extension"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/container"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var _ volumeModifier = EecModifier{}
+var _ volumeMountModifier = EecModifier{}
+var _ builder.Modifier = EecModifier{}
+
+const (
+	eecVolumeName = "eec-token"
+	eecMountPath  = "/var/lib/dynatrace/secrets/eec"
+	eecFile       = "token/eec.token"
+)
+
+func NewEecVolumeModifier(dk dynakube.DynaKube) EecModifier {
+	return EecModifier{
+		dk: dk,
+	}
+}
+
+type EecModifier struct {
+	dk dynakube.DynaKube
+}
+
+func (mod EecModifier) Enabled() bool {
+	return mod.dk.PrometheusEnabled()
+}
+
+func (mod EecModifier) Modify(sts *appsv1.StatefulSet) error {
+	baseContainer := container.FindContainerInPodSpec(&sts.Spec.Template.Spec, consts.ActiveGateContainerName)
+	sts.Spec.Template.Spec.Volumes = append(sts.Spec.Template.Spec.Volumes, mod.getVolumes()...)
+	baseContainer.VolumeMounts = append(baseContainer.VolumeMounts, mod.getVolumeMounts()...)
+
+	baseContainer.Command = []string{"/bin/sh"}
+	baseContainer.Args = []string{
+		"-c",
+		"cp -v " + eecMountPath + "/" + eecFile + " /var/lib/dynatrace/gateway/config/ ; /opt/dynatrace/gateway/entrypoint.sh",
+	}
```

Comment:
maybe dumb question, but this seems like to me a job for an init-container or even that we mount the secret with a `SubPath` directly into `/var/lib/dynatrace/gateway/config/`.

Overwriting the `Args` of the base container is asking for trouble, because  what if another `Modifier` wants to add something to it as well?
- **Adding** is fine (hopefully), but **overwriting** can cause unexpected behaviour (order of `Modifiers` will matter, they might overrule each other )

Reply:
[Done here](https://github.com/Dynatrace/dynatrace-operator/pull/3497/commits/94018fe4bea817f334f99fd309c8e6bb6130d5f5)

Reply:
At the end these lines are not needed anymore
[See here](https://github.com/Dynatrace/dynatrace-operator/pull/3497/commits/0c1242a98022fcb15b295175ab666b393fe79520)

Reply:
they were technically never needed anyway https://github.com/Dynatrace/dynatrace-operator/pull/3517 😛